### ISSUE TITLE
twistd should use twisted.logger

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8235]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8235|
|Type|enhancement|
|Created|2016-03-17 13:39:13Z|
|Branch|https://github.com/twisted/twisted/tree/tlogger-twistd-8235|

twisted.python.log is old, crusty, and is causing us problems (see [#6789](https://github.com/twisted/twisted/issues/6789)). We should use the new APIs when using twistd.

<details><summary>Searchable metadata</summary>

```
trac-id__8235 8235
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__New_Logging_System New Logging System
branch__branches_tlogger_twistd_8235 branches/tlogger-twistd-8235
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__logger logger
keywords__None None
time__1458221953854921 1458221953854921
changetime__1481165991180970 1481165991180970
version__None None
owner__wsanchez wsanchez
cc__wsanchez@...
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [47008]) Branching to tlogger-twistd-8235.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

Yeah, I'm happy with this.

Builds ran, all green, please review.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_3"></a>|@wsanchez set owner to @wsanchez<br>@wsanchez set status to `assigned`|
|-|-|

Wow, excellent.  I'll try to give this a look this afternoon.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_4"></a>|@wsanchez commented|
|-|-|

This is great.  Some minor feedback:

In `twisted.logger.__init__.py`:

 * I'd move `_logFor` to `twisted.logger._logger` and not put it into the module init, which should only contain exports from the private sub-modules.  I'd also call if `loggerFor`, but since it's private and not exported that's not terribly important.

 * `_` as a variable in the `_logFor` lambda expression is pretty confusing, because I expect that to be ignored, yet it's used twice.  Even `obj` is at least saying it's a thing being used.  I know this is just moving existing code, but it's small and it bugs me, so I had to mention it.

(I suspect we should make this function public at some point, but that's another ticket.)

In `twisted/application/app.py`:

 * I'm unsure of the process for deciding to deprecate something… so I wanna ask if we're all on board with it.

 * I think that it's more reasonable to deprecate only the case where an observer is defined that doesn't explicitly declare compliance with either of the new or old `ILogObserver` interfaces.  That is, I'd add a check for the old `ILogObserver` and not issue a warning in that case.  Note that both the new and old frameworks will work in observers that are simply callables (eg. `lambda event: events.append(event)`), but I think it's fair to expect an interface-declaring object in this case, especially since we have to default to assuming it's an old-style observer for backwards compatibility when one isn't explicit, and it would be easy to drop a new-style observer here and not notice it's being wrapped by `LegacyLogObserverWrapper`.

 * I wonder why `textFileLogObserver` is referenced as `logger.textFileLogObserver` when everything else in the `logger` module is imported explicitly; this seems unnecessarily inconsistent.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)|@wsanchez removed owner<br>@wsanchez set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_8"></a>|@hawkowl commented|
|-|-|

Thanks for the review, wsanchez.

I've addressed the comments, and the builders are spun, everything is looking green.

Since you didn't say whether to put up for review or merge after this, I'll err on the side of a second review.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)|@wsanchez set owner to @wsanchez<br>@wsanchez set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_10"></a>|@wsanchez commented|
|-|-|

Thanks for incorporating the feedback.

I did notice on a re-read that the doc string to `test_unmarkedObserversDeprecated` appears truncated.  Other than that, this looks good to merge.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_11"></a>|@hawkowl set status to `closed`|
|-|-|

(In [47084]) Merge tlogger-twistd-8235: Port twisted.application.app (and twistd) to use the new logging system.

Author: hawkowl
Reviewer: wsanchez
Fixes: [#8235](https://github.com/twisted/twisted/issues/8235)
